### PR TITLE
Add SVHN dataset

### PR DESCRIPTION
**Pull request by hhadian**
This is really odd that github lists previous commits.

### CODE REVIEW

**Comment by danpovey** [comment]
It's because I did squash and merge and you didn't rebase.  But it's OK, I'll squash this time too so list is irrelevant.

**Comment by danpovey** [comment]
Actually, why don't you rebase.  I notice the old changes are there, that shouldn't be, and it makes me nervous.

**Comment by hhadian** [comment]
I rebased just before my previous PR today (less than 2 hours ago)!
I will rebase everytime before making a PR from now on. I will rebase this too.

**Comment by hhadian** [comment]
Rebased it.

**Comment by danpovey** [comment]
It looks great, but I'm not sure that I want to be the first one to write example scripts for it.  Do you want to add a first guess at some training setup?

Incidentally, when dumping egs, you might want to aim for fewer archives, by using a larger number of images-per-archive (e.g. 50k).  I'm not sure how many archives it produces currently.  The training time per job is a little low right now, with the cifar setup.

**Comment by hhadian** [comment]
Actually (to test things) I adopted 1e (from cifar) and started training. It is currently at epoch 30 with accuracy ~95%. 
The training time per job for SVHN is currently ~22 seconds. I will try to put more images in archives.

**Comment by danpovey** [comment]
FYI: For ASR tasks we generally aim for about 3 to 5 minutes per job.
However we don't want to split the data into fewer archives than num-jobs-final.
And it could be that these models train fast enough that we want to do the averaging more frequently than that.
Also we don't really know what type of models we are going to end up using.
E.g. the wide-resnet configurations I am experimenting with are slower.

But we can always dump the egs in different sizes later if needed.

**Comment by hhadian** [comment]
I am not sure I understood the part about averaging. 
But currently 3 archives are created., which leads to 2N iterations (N is number of epochs).
If we make 1 archive instead of 3 and set num-jobs-initial and final to 1, then we will have N iterations.

If I understood things right, the only way to make it considerably faster (i.e do only N/5 iterations for e.g.) is to change the code to enable more than 1 epoch per iteration (through a flag like epochs_per_iter) by feeding an archive more than once to nnet3-train.
Then we can simply choose a suitable epochs_per_iter for each experiment (i.e. 1a, 1b, res1a, ...)

**Comment by danpovey** [comment]
@hhadian, I can merge this soon, but could you add at least one example script with numbers?

**Comment by danpovey** [comment]
actually, you focus on the CIFAR stuff; I'm running this setup and I can add an example script.

**Comment by hhadian** [comment]
I had 2 scripts from last week and I pushed them (before seeing your last message).

**Comment by danpovey** [comment]
it says I closed this; I didn't mean to.  I'd reopen, but the button is grayed out.
 I'm not seeing your example scripts, either from the github interface or when I check out your repo from the command line.

**Comment by danpovey** [comment]
can you please create a new PR?  It won't let me reopen this one.

**Comment by hhadian** [comment]
Will do now
